service/frontend/workflow_handler.go

```diff
@@ -3108,6 +3111,246 @@ func (wh *WorkflowHandler) validateStartWorkflowArgsForSchedule(
+			Namespace: request.Namespace,
+			Execution: &commonpb.WorkflowExecution{
+				WorkflowId: deploymentWorkflowID,
+			},
+			Query: &querypb.WorkflowQuery{QueryType: deployment.QueryDescribeDeployment},
+		},
+	}
+
+	res, err := wh.historyClient.QueryWorkflow(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	var queryResponse deployspb.DescribeResponse
+	err = payloads.Decode(res.GetResponse().GetQueryResult(), &queryResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	// build out task-queues for the response object
+	var taskQueues []*deploypb.DeploymentInfo_TaskQueueInfo
+	deploymentLocalState := queryResponse.DeploymentLocalState
+
+	for taskQueueName, taskQueueFamilyInfo := range deploymentLocalState.TaskQueueFamilies {
+		for _, taskQueueInfo := range taskQueueFamilyInfo.TaskQueues {
+			element := &deploypb.DeploymentInfo_TaskQueueInfo{
+				Name:            taskQueueName,
+				Type:            enumspb.TaskQueueType(taskQueueInfo.TaskQueueType),
+				FirstPollerTime: taskQueueInfo.FirstPollerTime,
+			}
+			taskQueues = append(taskQueues, element)
+		}
+	}
+
+	return &workflowservice.DescribeDeploymentResponse{
+		DeploymentInfo: &deploypb.DeploymentInfo{
+			Deployment:     deploymentLocalState.WorkerDeployment,
+			CreateTime:     deploymentLocalState.CreateTime,
+			TaskQueueInfos: taskQueues,
+			Metadata:       deploymentLocalState.Metadata,
+			IsCurrent:      deploymentLocalState.IsCurrentDeployment,
+		},
+	}, nil
+}
+
+func (wh *WorkflowHandler) GetCurrentDeployment(ctx context.Context, request *workflowservice.GetCurrentDeploymentRequest) (_ *workflowservice.GetCurrentDeploymentResponse, retError error) {
+	defer log.CapturePanic(wh.logger, &retError)
+
+	if request == nil {
+		return nil, errRequestNotSet
+	}
+
+	if len(request.Namespace) == 0 {
+		return nil, errNamespaceNotSet
+	}
+
+	if !wh.config.EnableDeployments(request.Namespace) {
+		return nil, errDeploymentsNotAllowed
+	}
+
+	namespaceID, err := wh.namespaceRegistry.GetNamespaceID(namespace.Name(request.GetNamespace()))
+	if err != nil {
+		return nil, err
+	}
+
+	// validating params
+	err = deployment.ValidateDeploymentWfParams(deployment.SeriesFieldName, request.SeriesName, wh.config.MaxIDLengthLimit())
+	if err != nil {
+		return nil, err
+	}
+
+	// Query the deployment series workflow to get the current deployment
+	workflowID := deployment.GenerateDeploymentSeriesWorkflowID(request.SeriesName)
+	req := &historyservice.QueryWorkflowRequest{
+		NamespaceId: namespaceID.String(),
+		Request: &workflowservice.QueryWorkflowRequest{
+			Namespace: request.Namespace,
+			Execution: &commonpb.WorkflowExecution{
+				WorkflowId: workflowID,
+			},
+			Query: &querypb.WorkflowQuery{QueryType: deployment.QueryCurrentDeployment},
+		},
+	}
+
+	res, err := wh.historyClient.QueryWorkflow(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	var currentDeployment string
+	err = payloads.Decode(res.GetResponse().GetQueryResult(), &currentDeployment)
```

Comment:
does the query return a build ID or deployment? if it's a build ID then the var name should change.